### DISCUSSION TITLE
登录问题：无法调用account脚本

### DISCUSSION THREAD

**Discussion by zhangyq73** [kbengine/kbengine#1379] [General]
kbe_accountinfos表中有玩家数据，但tbl_account表里没有；数据库数据是一样的，在linux环境下是可以登录的，在win10下是失败的：baseapp日志：
  DEBUG    baseapp01 34511 27000  [2021-02-24 15:34:22 275] - PendingLoginMgr::add: test022, size=1.
   DEBUG    baseapp01 34511 27000  [2021-02-24 15:34:22 344] - Channel::handshake: kcp(192.168.1.8:54557/1/0/0) successfully!
    INFO    baseapp01 34511 27000  [2021-02-24 15:34:22 535] - ServerApp::onHello: verInfo=2.5.8, scriptVerInfo=0.1.0, encryptedKey=None, addr:192.168.1.8:54557/1/0/0
 WARNING    baseapp01 34511 27000  [2021-02-24 15:34:22 535] - Baseapp::onHello: client is not encrypted, addr=192.168.1.8:54557/1/0/0
    INFO    baseapp01 34511 27000  [2021-02-24 15:34:22 734] - Baseapp::loginBaseapp: new user[test022], channel[192.168.1.8:54557/1/0/0].
   DEBUG    baseapp01 34511 27000  [2021-02-24 15:34:22 752] - PendingLoginMgr::remove: test022, size=0.
   ERROR    baseapp01 34511 27000  [2021-02-24 15:34:22 752] - Baseapp::onQueryAccountCBFromDbmgr: query test022 is failed! error()
    INFO    baseapp01 34511 27000  [2021-02-24 15:36:22 354] - ServerApp::onChannelTimeOut: Channel 192.168.1.8:54557/1/0/0 timeout!
dbmgr日志：
   DEBUG      dbmgr-1 34511 24000  [2021-02-24 15:33:37 319] - Dbmgr::DBTaskExecuteRawDatabaseCommand::presentMainThread: SELECT * FROM Hall_exchangeSell.
   DEBUG      dbmgr-1 34511 24000  [2021-02-24 15:33:37 319] - Dbmgr::DBTaskExecuteRawDatabaseCommand::presentMainThread: SELECT * FROM Hall_exchangeBeg.
  S_INFO      dbmgr-1 34511 24000  [2021-02-24 15:34:22 260] - test022
   DEBUG      dbmgr-1 34511 24000  [2021-02-24 15:34:22 273] - Dbmgr::onAccountLogin:loginName=test022, accountName=test022, success=0, componentID=0, entityID=0, dbid=66, flags=0, deadline=0.
  S_INFO      dbmgr-1 34511 24000  [2021-02-24 15:34:22 735] - test022
   DEBUG      dbmgr-1 34511 24000  [2021-02-24 15:34:22 751] - Dbmgr::queryAccount: test022, success=false, flags=0, deadline=0.
问题应该是在Baseapp::loginBaseapp: new user[test022]的时候并没调用account脚本，但linux和windows下脚本、kbe服务器版本号都一样

经过比对日志：正确的是PendingLoginMgr::remove以后，会调用EntityApp::createEntity: new Account；错误的没有直接再次查询执行Baseapp::onQueryAccountCBFromDbmgr，导致失败

[logs.zip](https://github.com/kbengine/kbengine/files/6034487/logs.zip)

**Comment by worldg** [comment]
tbl_account存储失败了？ 没有错误？

**Comment by zhangyq73** [comment]
tbl_account中没有对应的玩家数据，所以失败了。新用户没问题，linux下不知道为什么可以登陆进去，windows不行，同样是kbe_accountinfos有数据，但tbl_account表里没有。
